mne/transforms.py

```diff
@@ -1524,30 +1524,51 @@ def _euler_to_quat(euler):
     return quat
 
 
+def _validate_steps(param, validate_type=None, name=None):
+    """Validate that the param only contains accepted steps."""
+    for step in param:
+        if step not in ('trans', 'rotate', 'scale', 'sdr'):
+            raise TypeError(f"``step`` must be one of ``'trans', "
+                            "'rotate', 'scale', 'sdr'`` "
+                            f"got {step} instead")
+        if validate_type is not None:
+            _validate_type(param[step], validate_type, f"{name}['{step}'']")
+
+
 @verbose
-def compute_volume_registration(moving, static, zooms=None,
-                                niter_affine=(100, 100, 10),
-                                niter_sdr=None, rigid=False,
-                                verbose=None):
-    """Align two volumes using an affine and SDR.
+def compute_volume_registration(moving, static, steps='warp', zooms=None,
+                                niter=None, verbose=None):
+    """Align two volumes using an affine and, optionally, SDR.
 
     Parameters
     ----------
     %(moving)s
     %(static)s
+    steps : str | list
+        The registration steps to perform. The possible steps are translation
+        (``"trans"``), rotation (``"rotate"``), scaling (``"scale"``)
+        and symmetric diffeomorphic registration which is a non-linear
+        similarity-matching algorithm (``"sdr"``). If ``"warp"`` (default),
+        all the steps will be performed. If ``rigid``, only the translation
+        and rotation steps will be performed, which registers the volume
+        without distorting its underlying structure. Can be a list of steps
+        with any steps that are to be skipped omitted.
     zooms : float | tuple | dict | None
         The voxel size of volume for each spatial dimension in mm.
         If None (default), MRIs won't be resliced (slow, but most accurate).
         Can be a tuple to provide separate zooms for each dimension (X/Y/Z),
-        or a dict with keys ``"affine"`` and ``"sdr"`` (with values that are
+        or a dict with keys ``"scale"`` and ``"sdr"`` (with values that are
         float`, tuple, or None) to provide separate reslicing/accuracy for the
         affine and SDR morph steps.
-    %(niter_affine)s
-    %(niter_sdr)s
-        Can be None (default) to set ``niter_sdr=()`` automatically when
-        ``rigid=True``.
-    rigid : bool
-        If True, do a rigid alignment instead of a full affine transformation.
+    niter : dict | None
+        For each phase of the registration, ``niter`` is the number of
+        iterations per successive stage of optimization. The length of
+        ``niters`` defines the number of successive stages of optimization.
+        Default is ``(100, 100, 10)`` for the ``"trans"``, ``"rotate"`` and
+        ``"scale"`` steps and ``(5, 5, 3)`` for the ``"sdr"`` step. If a
```

Comment:
@alexrockhill -1 on changing this name to scale. IIUC it's not just a scaling, it's a full affine, which includes shear

Reply:
Sounds good, I was just throwing out a way to try and make it more accessible but I think you're right it's too simplified.